extensions/wikia/PageHeader/Subtitle.class.php

```diff
@@ -0,0 +1,265 @@
+		} else if ( defined( 'NS_BLOG_ARTICLE' ) && $this->title->getNamespace() === NS_BLOG_ARTICLE ) {
+			return $this->getBlogArticleSubtitle( $app );
+		}
+
+		return $output->getSubtitle();
+	}
+
+	private function getEditPageSubtitle() {
+		$wgOutput = RequestContext::getMain()->getOutput();
+
+		$subtitle = [
+			$this->getBackLink(),
+			$wgOutput->getSubtitle(),
+		];
+
+		if ( $this->request->getVal( 'action', 'view' ) === 'history' ) {
+			$sk = new FakeSkin();
+			$sk->setRelevantTitle( $this->title );
+
+			$undeleteLink = $sk->getUndeleteLink();
+			if ( $undeleteLink ) {
+				$subtitle[] = $undeleteLink;
+			}
+		}
+
+		$pipe = wfMessage( 'pipe-separator' )->escaped();
+
+		return implode( $pipe, $subtitle );
+	}
+
+	private function getTalkPageBackLink() {
+		if ( $this->title->isTalkPage() ) {
+			$namespace = $this->title->getNamespace();
+
+			// back to subject article link
+			switch ( $namespace ) {
+				case NS_TEMPLATE_TALK:
+					$msgKey = 'page-header-subtitle-back-to-template';
+					break;
+
+				case NS_MEDIAWIKI_TALK:
+					$msgKey = 'page-header-subtitle-back-to-mediawiki';
+					break;
+
+				case NS_CATEGORY_TALK:
+					$msgKey = 'page-header-subtitle-back-to-category';
+					break;
+
+				case NS_FILE_TALK:
+					$msgKey = 'page-header-subtitle-back-to-file';
+					break;
+
+				default:
+					$msgKey = 'page-header-subtitle-back-to-article';
+			}
+
+			return \Linker::link(
+				$this->title->getSubjectPage(),
+				wfMessage( $msgKey )->escaped(),
+				[ 'accesskey' => 'c' ]
+			);
+		}
+
+		return null;
+	}
+
+	private function getPageType() {
+		$namespace = $this->title->getNamespace();
+
+		$pageType = null;
+
+		if ( $namespace === NS_MEDIAWIKI ) {
+			$pageType = wfMessage( 'page-header-subtitle-mediawiki' )->escaped();
+		} else if ( $namespace === NS_TEMPLATE ) {
+			$pageType = wfMessage( 'page-header-subtitle-template' )->escaped();
+		} else if (
+			$namespace === NS_SPECIAL &&
+			!$this->title->isSpecial( 'Forum' ) &&
+			!$this->title->isSpecial( 'ThemeDesignerPreview' )
+		) {
+			$pageType = wfMessage( 'page-header-subtitle-special' )->escaped();
+		} else if ( $namespace === NS_CATEGORY ) {
+			$pageType = wfMessage( 'page-header-subtitle-category' )->escaped();
+		} else if ( $namespace === NS_FORUM ) {
+			$pageType = wfMessage( 'page-header-subtitle-forum' )->escaped();
+		} else if ( defined( 'NS_BLOG_LISTING' ) && $namespace === NS_BLOG_LISTING ) {
+			$pageType = wfMessage( 'page-header-subtitle-blog-category' )->escaped();
+		}
+
+		return $pageType;
+	}
+
+	/**
+	 * @return array
+	 */
+	private function languageVariants() {
+		$variants = $this->skinTemplate->get( 'content_navigation' )['variants'];
+		if ( !empty( $variants ) ) {
+			return array_map( function ( $variant ) {
+				return Html::element( 'a', [
+					'href' => $variant['href'],
+					'rel' => 'nofollow',
+					'id' => $variant['id'],
+				], $variant['text'] );
+			}, $variants );
+		}
+
+		return [];
+	}
+
+	// FIXME
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Take the required action to fix the issue indicated by this "FIXME" comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1134)